### ISSUE TITLE
Export/Import projects button

### ISSUE THREAD

**Issue by TheJaredWilcurt** [TheJaredWilcurt/UGUI-pa11y#32]
Likely under the `File > Export` menu. This would export out your project's `settings.json` file to your desktop, so it can be shared with someone else. During the import process, the only machine-specific setting would be the output directory for where to store reports. So we'd need to do an `fs.exists` on the computer that is importing the settings, and if the output folder does not exist, then it will default to the user's desktop.

Maybe we need a UI for this, a modal that gives the user import/export tabs. Import settings would include what the default output folder should be, and if you want to add these projects (show a list of them), to your current projects, or replace your current projects with them? or maybe just always add, and if they want to remove stuff they can remove pages/projects manually? Need UX discussions.

Export modal could have a list of all projects, so you can check/uncheck what you want to export/share.

**Comment by ryanparrish** [comment]
[Import projects.pdf](https://github.com/open-indy/Koa11y/files/1331343/Import.projects.pdf)

This is what I was thinking about your modal window

**Comment by TheJaredWilcurt** [comment]
<p align="center"><img alt="import-export" src="https://user-images.githubusercontent.com/4629794/30888584-45da0f78-a2ef-11e7-9b08-8a6f450e43a5.png"></p>

Here's how I had envisioned it:

<p align="center"><img alt="Export wireframe" src="https://user-images.githubusercontent.com/4629794/30888396-5edf4a84-a2ee-11e7-9c84-fe7d7971561f.png"></p>

Since we control the output and input, we decide the format. And since the code is written in JavaScript, the format will be JSON, as it is easy to work with. So we don't give the user a choice here. The data is only to be used in Koa11y.

I want to give the user control over what projects they export, in case there are sites that are not relevant to who they would be sending the settings to. We may also want to add an option to "include actions", since the actions could contain passwords that they don't want to share as part of the export.

In your design you can't export more than one project at a time, which is an arbitrary restriction caused by the UI, not the technology or user needs.

Alternatively, we could list all projects/pages in a tabular manner and have the required, actions, image alts, etc be individual checkboxes for each item.

For imports the screen would initially look like this:

<p align="center"><img alt="Import wireframe" src="https://user-images.githubusercontent.com/4629794/30888405-60f9f5ee-a2ee-11e7-8784-ca90fda15d9d.png"></p>

Then after a file is imported, it would change to look similar to the Export tab, and have a file tree for them to choose what to import and the "Import" button would be enabled.

Also I forgot to put an X in the corner of my modals, but there should be one.

**Comment by ryanparrish** [comment]
okay thank you for showing me your process

**Comment by ryanparrish** [comment]
17-10-05 

We talked about Koa11y export screen options discussed: 

Per project share button 
select box drag and drop project box. 
tab for each project. 
Combo box for each box. 
![kwctx](https://user-images.githubusercontent.com/8293865/31255237-d9b4199e-a9f9-11e7-90ad-70c1f2adec2d.png)
